ckanext/privatedatasets/actions.py

```diff
@@ -194,3 +194,110 @@ def acquisitions_list(context, data_dict):
+                # if a list of allowed users is included in a public dataset. However, this check
+                # should be performed in order to avoid strange future exceptions
+                if dataset.get('private', None) is True:
+
+                    # Create the array if it does not exist
+                    if constants.ALLOWED_USERS not in dataset or dataset[constants.ALLOWED_USERS] is None:
+                        dataset[constants.ALLOWED_USERS] = []
+
+                    # Deletes the user only if it is in the list
+                    if user_info['user'] in dataset[constants.ALLOWED_USERS]:
+                        dataset[constants.ALLOWED_USERS].remove(user_info['user'])
+                        context_pkg_update = context.copy()
+                        context_pkg_update['ignore_auth'] = True
+
+                        # Set creator as the user who is performing the changes
+                        user_show = plugins.toolkit.get_action('user_show')
+                        creator_user_id = dataset.get('creator_user_id', '')
+                        user_show_context = {'ignore_auth': True}
+                        user = user_show(user_show_context, {'id': creator_user_id})
+                        context_pkg_update['user'] = user.get('name', '')
+
+                        plugins.toolkit.get_action('package_update')(context_pkg_update, dataset)
+                        log.info('Allowed Users deleted correctly')
+                    else:
+                        log.warn('The user %s is already deleted from accessing the %s dataset' % (user_info['user'], dataset_id))
+                else:
+                    log.warn('Dataset %s is public. Allowed Users cannot be deleted')
+                    warns.append('Unable to upload the dataset %s: It\'s a public dataset' % dataset_id)
+
+            except plugins.toolkit.ObjectNotFound:
+                # If a dataset does not exist in the instance, an error message will be returned to the user.
+                # However the process won't stop and the process will continue with the remaining datasets.
+                log.warn('Dataset %s was not found in this instance' % dataset_id)
+                warns.append('Dataset %s was not found in this instance' % dataset_id)
+            except plugins.toolkit.ValidationError as e:
+                # Some datasets does not allow to introduce the list of allowed users since this property is
+                # only valid for private datasets outside an organization. In this case, a wanr will return
+                # but the process will continue
+                # WARN: This exception should not be risen anymore since public datasets are not updated.
+                message = '%s(%s): %s' % (dataset_id, constants.ALLOWED_USERS, e.error_dict[constants.ALLOWED_USERS][0])
+                log.warn(message)
+                warns.append(message)
```

Comment:
Every time a developer copies and pastes a 100 line method just for changing a single line (the 268 in this case), god kills a kitty :-(